homeassistant/components/delijn/sensor.py

```diff
@@ -37,67 +42,63 @@
-        line = Passages(
-            hass.loop, stop_id, number_of_departures, api_key, session, True
+        sensors.append(
+            DeLijnPublicTransportSensor(
+                Passages(
+                    hass.loop,
+                    nextpassage[CONF_STOP_ID],
+                    nextpassage[CONF_NUMBER_OF_DEPARTURES],
+                    api_key,
+                    session,
+                    True,
+                )
+            )
         )
-        await line.get_passages()
-        if line.passages is None:
-            _LOGGER.warning("No data received from De Lijn")
-            return
-        sensors.append(DeLijnPublicTransportSensor(line, name))
 
     async_add_entities(sensors, True)
 
 
 class DeLijnPublicTransportSensor(Entity):
     """Representation of a Ruter sensor."""
 
-    def __init__(self, line, name):
+    def __init__(self, line):
         """Initialize the sensor."""
         self.line = line
         self._attributes = {ATTR_ATTRIBUTION: ATTRIBUTION}
-        self._name = name
+        self._name = None
         self._state = None
-        self._available = False
 
     async def async_update(self):
         """Get the latest data from the De Lijn API."""
-        await self.line.get_passages()
-        if self.line.passages is None:
-            _LOGGER.warning("No data received from De Lijn")
+        try:
+            await self.line.get_passages()
+            self._name = await self.line.get_stopname()
+        except HttpException:
+            self._state = STATE_UNAVAILABLE
+            _LOGGER.error("De Lijn http error")
             return
+
         try:
             first = self.line.passages[0]
             if first["due_at_realtime"] is not None:
                 first_passage = first["due_at_realtime"]
             else:
                 first_passage = first["due_at_schedule"]
             self._state = first_passage
-            self._name = first["stopname"]
-            self._attributes["stopname"] = first["stopname"]
             self._attributes["line_number_public"] = first["line_number_public"]
             self._attributes["line_transport_type"] = first["line_transport_type"]
             self._attributes["final_destination"] = first["final_destination"]
             self._attributes["due_at_schedule"] = first["due_at_schedule"]
             self._attributes["due_at_realtime"] = first["due_at_realtime"]
             self._attributes["next_passages"] = self.line.passages
-            self._available = True
-        except (KeyError, IndexError) as error:
-            _LOGGER.debug("Error getting data from De Lijn: %s", error)
-            self._available = False
-
-    @property
-    def available(self):
-        """Return True if entity is available."""
-        return self._available
+        except (KeyError, IndexError):
+            _LOGGER.error("Invalid data received from De Lijn")
+            self._state = STATE_UNAVAILABLE
```

Comment:
We should implement the `available` property like before. Don't remove that.

Reply:
Don't know how that slipped through, I put it back.